capstone/src/main/java/com/google/sps/Objects/Comment.java

```diff
@@ -22,6 +22,24 @@ public static Comment getCommentEntity(EmbeddedEntity entity) {
     return new Comment(timestamp, commentText, userId);
   }
 
+  public static EmbeddedEntity toEntity(String commentText, String userId) {
+    EmbeddedEntity commentEntity = new EmbeddedEntity();
+    commentEntity.setProperty("timestamp", System.currentTimeMillis());
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("userId", userId);
+    return commentEntity;
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (other == null) return false;
+    if (other == this) return true;
+    if (!(other instanceof Comment)) return false;
+    Comment comment = (Comment) other;
+    return commentText.equals(comment.commentText) &&
+      userId.equals(comment.userId);
```

Comment:
@lxcyqx @himaniyadav

I would challenge this equality check, and others. If I comment twice on two different challenges or even on the same challenge with the same text, this equality check is claiming they're the same. I don't know exactly how equals plans to be used throughout the entire server, but I could see equals doing one of two things:

1) Just check if the object ID's are the same. The benefit of this is that if I post something and retrieve it later from the database after somebody has liked it, we can still call these posts as "equal" even though not all fields are up to date. Downside is you guys will have to add some additional logic around to/from entity to better manage ID's.
2) Check the Object ID is the same or that the timestamp is the same, as well as all other fields. The downside is what you're already seeing, that you have to check all fields manually and update it as you go. The use case here is that you'd now have 2 kinds of equals: obj1.id === obj2.id and obj1.equals(obj2) which would be used for different cases.

EITHER WAY I think that a lot of what I'm seeing in the equality checks may be completely invalidated without using something like an ID or timestamp to actually check for uniqueness.

Reply:
I'm using the equality checks more as a way to make sure the correct data is being posted/retrieved, but not necessarily checking for uniqueness. Since the timestamp and object ID's are being generated when an entity is created/put in the database, I haven't found a way to mandate that. Could I add a test to make sure two objects with the exact same data have two different ID's for uniqueness?

Reply:
If you're checking for data retrieval correctness, my argument would be that this logic should exist in the unit tests instead of a production-available function then. Obviously if you're using `.equals()` somewhere in production you can keep the logic here, but your comment above seems more like unit-test logic than a method that should be usable in production.